### ISSUE TITLE
libc.so (__futex_wait_ex(void volatile*, bool, int, bool, timespec const*)+144)

### ISSUE THREAD

**Issue by centy** [flutter/flutter#158425]
### Steps to reproduce

Random crash

### Expected results

do not crash

### Actual results

app crashes

### Code sample

<details open><summary>Code sample</summary>

```dart
No sample
```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>

### Logs

<details open><summary>Logs</summary>

```console
#00  pc 0x00000000000870cc  /apex/com.android.runtime/lib64/bionic/libc.so (syscall+28)
  #01  pc 0x000000000008adb8  /apex/com.android.runtime/lib64/bionic/libc.so (__futex_wait_ex(void volatile*, bool, int, bool, timespec const*)+144)
Ta procedura synchronizacji natywnej blokuje wątek główny, prowadząc do błędu ANR. [Więcej informacji](https://developer.android.com/topic/performance/anrs/find-unresponsive-thread#lock-contention)

  #02  pc 0x00000000000ebccc  /apex/com.android.runtime/lib64/bionic/libc.so (pthread_cond_wait+60)
  #03  pc 0x000000000046ed98  /data/app/~~jOXQ91ZLWDZL4S-87fDGrA==/com.gst.color.numbers.pixyfy-rwED9wcRdhpGejQaA-3fzw==/split_config.arm64_v8a.apk!libflutter.so
  #04  pc 0x0000000000482a50  /data/app/~~jOXQ91ZLWDZL4S-87fDGrA==/com.gst.color.numbers.pixyfy-rwED9wcRdhpGejQaA-3fzw==/split_config.arm64_v8a.apk!libflutter.so
  #05  pc 0x00000000007c85a4  /data/app/~~jOXQ91ZLWDZL4S-87fDGrA==/com.gst.color.numbers.pixyfy-rwED9wcRdhpGejQaA-3fzw==/split_config.arm64_v8a.apk!libflutter.so
  #06  pc 0x00000000004671d4  /data/app/~~jOXQ91ZLWDZL4S-87fDGrA==/com.gst.color.numbers.pixyfy-rwED9wcRdhpGejQaA-3fzw==/split_config.arm64_v8a.apk!libflutter.so
  #07  pc 0x00000000004698ec  /data/app/~~jOXQ91ZLWDZL4S-87fDGrA==/com.gst.color.numbers.pixyfy-rwED9wcRdhpGejQaA-3fzw==/split_config.arm64_v8a.apk!libflutter.so
  at io.flutter.embedding.engine.FlutterJNI.nativeSurfaceDestroyed (FlutterJNI.java)
  at io.flutter.embedding.engine.FlutterJNI.onSurfaceDestroyed (FlutterJNI.java:667)
  at io.flutter.embedding.engine.renderer.FlutterRenderer.stopRenderingToSurface (FlutterRenderer.java:1125)
  at io.flutter.embedding.android.FlutterSurfaceView.disconnectSurfaceFromRenderer (FlutterSurfaceView.java:303)
  at io.flutter.embedding.android.FlutterSurfaceView.access$400 (FlutterSurfaceView.java:36)
  at io.flutter.embedding.android.FlutterSurfaceView$1.surfaceDestroyed (FlutterSurfaceView.java:78)
  at android.view.SurfaceView.notifySurfaceDestroyed (SurfaceView.java:1947)
  at android.view.SurfaceView.updateSurface (SurfaceView.java:1239)
  at android.view.SurfaceView.setWindowStopped (SurfaceView.java:306)
  at android.view.SurfaceView.surfaceDestroyed (SurfaceView.java:1771)
  at android.view.ViewRootImpl.notifySurfaceDestroyed (ViewRootImpl.java:1929)
  at android.view.ViewRootImpl.setWindowStopped (ViewRootImpl.java:1864)
  at android.view.WindowManagerGlobal.setStoppedState (WindowManagerGlobal.java:700)
  at android.app.Activity.performStop (Activity.java:8350)
  at android.app.ActivityThread.callActivityOnStop (ActivityThread.java:4964)
  at android.app.ActivityThread.performStopActivityInner (ActivityThread.java:4943)
  at android.app.ActivityThread.handleStopActivity (ActivityThread.java:5017)
  at android.app.servertransaction.StopActivityItem.execute (StopActivityItem.java:40)
  at android.app.servertransaction.TransactionExecutor.executeLifecycleState (TransactionExecutor.java:176)
  at android.app.servertransaction.TransactionExecutor.execute (TransactionExecutor.java:97)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:2142)
  at android.os.Handler.dispatchMessage (Handler.java:106)
  at android.os.Looper.loop (Looper.java:236)
  at android.app.ActivityThread.main (ActivityThread.java:7864)
  at java.lang.reflect.Method.invoke (Native method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:620)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1011)
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
(flutter: 3.27.0-0.1.pre)
```

</details>

**Comment by darshankawar** [comment]
@centy 
Is your app impeller enabled ? 
Looking at the stacktrace, it looks similar to https://github.com/flutter/flutter/issues/156728 which was fixed recently.  See if it resonates with your code implementation or use case.

**Comment by huycozy** [comment]
@centy I see the commit https://github.com/flutter/engine/commit/e2188ee9ab77266e68552a6a62cbbe5c9159053d that fixed for https://github.com/flutter/flutter/issues/156728 is available on Flutter master channel only, not on beta or stable yet. Please try on master to see if the crash persists or not.

**Comment by centy** [comment]
> @centy I see the commit [flutter/engine@e2188ee](https://github.com/flutter/engine/commit/e2188ee9ab77266e68552a6a62cbbe5c9159053d) that fixed for #156728 is available on Flutter master channel only, not on beta or stable yet. Please try on master to see if the crash persists or not.

I can not test is on master, we can close it, when it will be merged down to beta or stable I will test it.
